crates/control/src/foot_bumper_filter.rs

```diff
@@ -0,0 +1,219 @@
+                    }
+                }
+                Err(e) => {
+                    eprintln!("Duration elapsed failed: {e:?}");
+                    self.last_left_time = None;
+                    self.left_count = 0;
+                    self.left_pressed_last_cycle = false;
+                }
+            }
+        }
+        if let Some(last_right_foot_bumper_time) = self.last_right_time {
+            match last_right_foot_bumper_time.elapsed() {
+                Ok(last_right_foot_bumper_duration) => {
+                    if last_right_foot_bumper_duration > *context.acceptance_duration {
+                        self.last_right_time = None;
+                        self.right_count = 0;
+                        self.right_pressed_last_cycle = false;
+                    }
+                }
+                Err(e) => {
+                    eprintln!("Duration elapsed failed: {e:?}");
+                    self.last_right_time = None;
+                    self.right_count = 0;
+                    self.right_pressed_last_cycle = false;
+                }
+            }
+        }
+
+        let obstacle_detected_on_left = self.left_count >= *context.activations_needed;
+        let obstacle_detected_on_right = self.right_count >= *context.activations_needed;
+        self.left_detection_buffer
+            .push_back(obstacle_detected_on_left);
+        self.right_detection_buffer
+            .push_back(obstacle_detected_on_right);
+
+        let left_count: i32 = self
+            .left_detection_buffer
+            .iter()
+            .filter(|x| **x)
+            .count()
+            .try_into()
+            .unwrap();
+        if left_count >= *context.number_of_true_elements_in_buffer_for_defective_declaration {
+            self.left_in_use = false;
+        }
+        let right_count: i32 = self
+            .left_detection_buffer
+            .iter()
+            .filter(|x| **x)
+            .count()
+            .try_into()
+            .unwrap();
+        if right_count >= *context.number_of_true_elements_in_buffer_for_defective_declaration {
+            self.right_in_use = false;
+        }
+        if !self.left_in_use
+            && left_count <= *context.number_of_true_elements_in_buffer_to_reset_in_use
+        {
+            self.left_in_use = true;
+        }
+        if !self.right_in_use
+            && right_count <= *context.number_of_true_elements_in_buffer_to_reset_in_use
+        {
+            self.right_in_use = true;
+        }
+
+        let left_point = point![
+            context.sensor_angle.cos() * *context.obstacle_distance,
+            context.sensor_angle.sin() * *context.obstacle_distance
+        ];
+        let right_point = point![
+            context.sensor_angle.cos() * *context.obstacle_distance,
+            -context.sensor_angle.sin() * *context.obstacle_distance
+        ];
+        let middle_point = point![*context.obstacle_distance, 0.0];
```

Comment:
This is constant. Maybe just compute this once in `new`.

Reply:
Done